sdk/core/azure-core/src/main/java/com/azure/core/util/FluxUtil.java

```diff
@@ -83,6 +84,74 @@ private static void accept(ByteArrayOutputStream byteOutputStream, ByteBuffer by
         return byteArray;
     }
 
+    /**
+     * Converts an {@link InputStream} into a {@link Flux} of {@link ByteBuffer} using a chunk size of 4096.
+     * <p>
+     * Given that {@link InputStream} is not guaranteed to be replayable the returned {@link Flux} should be considered
+     * non-replayable as well.
+     * <p>
+     * If the passed {@link InputStream} is {@code null} {@link Flux#empty()} will be returned.
+     *
+     * @param inputStream The {@link InputStream} to convert into a {@link Flux}.
+     * @return A {@link Flux} of {@link ByteBuffer ByteBuffers} that contains the contents of the stream.
+     */
+    public static Flux<ByteBuffer> toFluxByteBuffer(InputStream inputStream) {
+        return toFluxByteBuffer(inputStream, 4096);
+    }
+
+    /**
+     * Converts an {@link InputStream} into a {@link Flux} of {@link ByteBuffer}.
+     * <p>
+     * Given that {@link InputStream} is not guaranteed to be replayable the returned {@link Flux} should be considered
+     * non-replayable as well.
+     * <p>
```

Comment:
Should we document that the caller is responsible for closing the provided stream? (if the user provides the produced Flux to our async API then they should dispose it once async API completes)

Reply:
Good question, I don't think in any other location where we are converting an `InputStream` into a `Flux<ByteBuffer>` we close the stream.

Reply:
yes, the Flux.generate is not closing stream and it shouldn't. I mean consumer of this API is in charge of dealing with it.  

```java
InputStream is = // open-file
FormRecognizer.someAsyncAPI(FluxUtils.toByteBufferFlux(is))
    .doFinally(s -> {
       is.close();
    }).block();
```

Reply:
The lifecyle of the inputstream should be managed by the user. They might decide to reset and reuse the inputstream if they want to. So, we shouldn't be closing user provided stream.

Reply:
yes.

Also, any thoughts on the replayability of such Fluxes?  I see this https://docs.oracle.com/javase/7/docs/api/java/io/InputStream.html#markSupported() API to check the stream can be marked and reset. Not sure how feasible it with File streams.

In FR APIs we do two subscriptions - one for contentType detection and another one when sending the content. The sync API takes `InputStream` from the user, convert that to Flux and call async API, so those APIs won't work as expected.

Reply:
@anuchandy I think it would be safer for the FormRecognizer API to not subscribe twice in case the Flux being sent isn't replayable, this goes back to the problem where we need to assume `Flux` isn't replayable since treating as non-replayable will work for replayable but not the other way around.

Reply:
I thought, in general, our SDKs assume Flux is "replayable" :), otherwise, retry is not going to work (retry requires multi-subscription). Storage is a special case where they are committed to supporting "non-replyable" Flux, that makes sense because of server-side chunk support.

We had some discussion on the same topic, in this thread https://github.com/Azure/azure-sdk-for-java/pull/9953#discussion_r411589700